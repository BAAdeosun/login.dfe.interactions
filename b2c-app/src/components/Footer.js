import React from 'react'

export default function footer() {
    return (
        <div>
            I AM THE FOOTER
        </div>
    )
}
