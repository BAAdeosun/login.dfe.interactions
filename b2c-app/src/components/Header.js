import React from 'react'

function Header() {
    return (
        <header class="govuk-header " role="banner" data-module="govuk-header">
            <div class="govuk-header__container govuk-width-container">
                <div class="govuk-header__content">
                <a href="#" class="govuk-header__link govuk-header__link--service-name">
                    National Careers Service
                </a>
                <button type="button" class="govuk-header__menu-button govuk-js-header-toggle" aria-controls="navigation" aria-label="Show or hide Top Level Navigation">Menu</button>
                <nav>
                    <ul id="navigation" class="govuk-header__navigation " aria-label="Top Level Navigation">
                    <li class="govuk-header__navigation-item govuk-header__navigation-item--active">
                        <a class="govuk-header__link" href="#1">
                        Job profiles
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#2">
                        Skills health check
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#3">
                        Find a course
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#4">
                        Contact an adviser
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#4">
                        About us
                        </a>
                    </li>
                    </ul>
                </nav>
                </div>
            </div>
        </header>
    )
}

export default Header;