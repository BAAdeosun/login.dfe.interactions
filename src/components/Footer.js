import Link from 'next/link'

const Footer = () => (
    <footer>
        This is the footer
    </footer>
)

export default Footer