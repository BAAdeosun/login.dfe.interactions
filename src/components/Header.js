import Link from 'next/link'

const Header = () => (
    <header>
        This is the header
    </header>
)

export default Header