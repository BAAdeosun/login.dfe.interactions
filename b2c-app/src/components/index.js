import Header from './Header/Header';
import Footer from './Footer/Footer';
import B2C from './B2C/B2C';
import HeaderTitle from './HeaderTitle/HeaderTitle';
import PageNavigation from './PageNavigation/PageNavigation';
import NavigationLink from './NavigationLink/NavigationLink';
import Breadcrumbs from './Breadcrumbs/Breadcrumbs';
import MainContainer from './MainContainer/MainContainer';
import PageTitle from './PageTitle/PageTitle';
import FooterNavigation from './FooterNavigation/FooterNavigation';
import FooterLink from './FooterLink/FooterLink';
import PasswordHelpContainer from './PasswordHelp/PasswordHelpContainer';
import PasswordHelp from './PasswordHelp/PasswordHelp';
import PageLevelErrorContainer from './PageLevelErrorContainer/PageLevelErrorContainer';
import CreateNewPassword from './CreateNewPassword/CreateNewPassword';
import DateOfBirth from './DateOfBirth/DateOfBirth';
import TermsAndConditions from './TermsAndConditions/TermsAndConditions';
import ButtonLink from './ButtonLink/ButtonLink';
import Link from './Link/Link';
import Paragraph from './Paragraph/Paragraph';
import Postcode from './Postcode/Postcode';
import InputField from './InputField/InputField';

export default {
    Header: Header,
    Footer: Footer,
    B2C: B2C,
    HeaderTitle: HeaderTitle,
    PageNavigation: PageNavigation,
    NavigationLink: NavigationLink,
    Breadcrumbs: Breadcrumbs,
    MainContainer: MainContainer,
    PageTitle: PageTitle,
    FooterNavigation: FooterNavigation,
    FooterLink: FooterLink,
    PasswordHelpContainer: PasswordHelpContainer,
    PasswordHelp: PasswordHelp,
    PageLevelErrorContainer: PageLevelErrorContainer,
    CreateNewPassword: CreateNewPassword,
    DateOfBirth: DateOfBirth,
    TermsAndConditions: TermsAndConditions,
    ButtonLink: ButtonLink,
    Link: Link,
    Paragraph: Paragraph,
    Postcode: Postcode,
    InputField: InputField
}