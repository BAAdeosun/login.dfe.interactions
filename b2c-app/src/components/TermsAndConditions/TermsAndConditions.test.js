import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import components from '..';

const testErrors = {
    tsAndCs: {
        text: 'this is a test'
    }
};
const updateParentMock = jest.fn();

it('renders without crashing', () => {
    shallow(<components.TermsAndConditions />);
});

it('renders correctly without props passed in', () => {
    const tree = renderer
        .create(<components.TermsAndConditions />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly with errors', () => {
    const tree = renderer.create(<components.TermsAndConditions showErrors={true} visibleErrors={testErrors} />);
    tree.root.instance.setState({
        errors: {
            tsAndCs: {
                visible: {
                    text: 'this is a test'
                }
            }
        }
    });
    expect(tree.toJSON()).toMatchSnapshot()
});

it('calls validation, sets errors and calls onChange callback', () => {

    const mockOnChangeCallback = jest.fn();

    const wrapper = mount(
        <components.TermsAndConditions
            showErrors={true}
            onChange={mockOnChangeCallback}
            visibleErrors={testErrors}
            updateParentErrors={updateParentMock} />
    );
    const tsAndCsCheckbox = wrapper.find('#tsAndCsCustom');
    const validationSpy = jest.spyOn(wrapper.instance(), 'isValidTsAndCs');
    let changeEvent = {
        target:
        {
            name: 'tsAndCsAccepted',
            checked: true
        }
    };


    //default values set
    expect(wrapper.state().tsAndCsAccepted).toBe(false);
    expect(wrapper.state().errors.tsAndCs.text).toEqual('You must accept our Terms and Conditions');

    //simulate checking Ts & Cs
    tsAndCsCheckbox.simulate('change', changeEvent);
    //validation called
    expect(validationSpy).toHaveBeenCalled();
    //state value has been updated
    expect(wrapper.state().tsAndCsAccepted).toBe(true);
    //check error message is empty
    expect(wrapper.state().errors.tsAndCs.text).toEqual('');
    //callback function passed in has been called
    expect(mockOnChangeCallback).toHaveBeenCalledWith({ tsAndCsAccepted: true });

    //simulate unchecking Ts & Cs
    changeEvent.target.checked = false;
    tsAndCsCheckbox.simulate('change', changeEvent);
    //validation called
    expect(validationSpy).toHaveBeenCalled();
    //state value has been updated
    expect(wrapper.state().tsAndCsAccepted).toBe(false);
    //callback function passed in has been called
    expect(mockOnChangeCallback).toHaveBeenCalledWith({ tsAndCsAccepted: false });
    //check error message is empty
    expect(wrapper.state().errors.tsAndCs.text).toEqual('You must accept our Terms and Conditions');

});