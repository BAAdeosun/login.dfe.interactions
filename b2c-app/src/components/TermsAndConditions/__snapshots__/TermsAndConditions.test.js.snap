// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly without props passed in 1`] = `
<div>
  <h1
    className="govuk-heading-m"
  >
    Terms and conditions
  </h1>
  <div
    className="govuk-form-group "
  >
    <label
      className="block-label"
      htmlFor="tsAndCsCustom"
    >
      <input
        aria-invalid="true"
        checked={false}
        id="tsAndCsCustom"
        name="tsAndCsAccepted"
        noValidate={true}
        onChange={[Function]}
        type="checkbox"
        value={true}
      />
      I accept the 
      <a
        href="https://nationalcareers.service.gov.uk/help/terms-and-conditions"
        id="tsAndCsLink"
        rel="noopener noreferrer"
        target="_blank"
      >
        terms and conditions
      </a>
       and I am 13 or over
    </label>
  </div>
</div>
`;
